### PR TITLE
Issue 82 backend tests

### PR DESCRIPTION
**Pull request by TWoerner94**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,6 +30,15 @@
...
-    "concurrently": "^3.5.1"
+    "chai": "^4.1.2",
+    "chai-spies": "^1.0.0",
+    "concurrently": "^3.5.1",
+    "eslint-config-airbnb": "^16.1.0",
+    "eslint-plugin-import": "^2.9.0",
+    "eslint-plugin-jsx-a11y": "^6.0.3",
+    "eslint-plugin-react": "^7.7.0",
+    "mocha": "^5.0.5",
+    "mock-socket": "^7.1.0",
```
Comment (martiL) [inline package.json]: do you need mock-socket ?

File: server.js [hunk 2]
```diff
@@ -17,105 +14,93 @@ app.set("port", process.env.PORT || 3001);
...
-    const response = e.data;
-    res.status(200).send(response);
-  }
-
-  socket.send(obj);
+  sf.threeDTrafoSendToSocket(req.body.coords, (response, isOk) => {
+    if (isOk) {
+      res.status(200).send(response);
+    } else {
+      res.status(500).send(response);
```
Comment (martiL) [inline server.js]: write an issue for handling unknown errors. there should some kind of general error callback

File: socketFunctions/socketFunctions.js [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+const threeDTrafoDifferenceSendToSocket = (params, callback) => {
+  const startPoints = params.startPoints;
+  const targetPoints = params.targetPoints;
+  const trafoParams = params.trafoParams;
+
+  let calculatedTarget = [];
+  let differences = [];
+
+  let socket = getWebSocket().then(function(socket) {
+    startPoints.map((point, i) => {
```
Comment (brauls) [inline socketFunctions/socketFunctions.js]: maybe here would a forEach be more appropriate, cause you are not mapping a list to another list, but execute some code for each item in a list

File: socketFunctions/socketFunctions.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    }
+
+    socket.send(threeDTrafoRequest);
+  }).catch(function(err) {
+    callback(err, false)
+  });
+}
+
+const threeDTrafoDifferenceSendToSocket = (params, callback) => {
+  const startPoints = params.startPoints;
```
Comment (martiL) [inline socketFunctions/socketFunctions.js]: globalIdCounter is missing ?

**Pull request merged**